validate_helm_chart.sh

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env bash
+#
+# Validate that the image tags in the Helm chart match the chart version.
+#
+
+set -e
+shopt -s expand_aliases
+
+DOCKER=${DOCKER:-docker}
+
+helm() {
+  "${DOCKER}" run --user "$(id -u):$(id -g)" --rm -v "$(pwd)":/apps alpine/helm:3.15.1 "$@"
+}
+
+yq () {
+  "${DOCKER}" run -i --user "$(id -u):$(id -g)" --rm -v "${PWD}":/workdir mikefarah/yq:4.40.5 "$@"
+}
+
+usage() {
+    >&2 echo "usage: $0 <chart-tgz-file>"
+    >&2 echo
+    >&2 echo "example: $0 cilium-1.15.5.tgz"
+    >&2 echo "example: $0 tetragon-1.1.0.tgz"
+}
+
+CILIUM_IMAGE_PATHS=(
+  '{$.clustermesh.apiserver.image.tag}'
+  '{$.hubble.relay.image.tag}'
+  '{$.image.tag}'
+  '{$.operator.image.tag}'
+  '{$.preflight.image.tag}'
+)
```

Comment:
might become a bit painful to manage this list if we add more images in the future but i wasn't sure what else to do. anyways i validated it works with v1.13, v1.14, and v1.15:

```
% ./validate_helm_chart.sh cilium-1.15.5.tgz
SUCCESS: cilium {$.clustermesh.apiserver.image.tag}=v1.15.5 matches chart version 1.15.5
SUCCESS: cilium {$.hubble.relay.image.tag}=v1.15.5 matches chart version 1.15.5
SUCCESS: cilium {$.image.tag}=v1.15.5 matches chart version 1.15.5
SUCCESS: cilium {$.operator.image.tag}=v1.15.5 matches chart version 1.15.5
SUCCESS: cilium {$.preflight.image.tag}=v1.15.5 matches chart version 1.15.5

% ./validate_helm_chart.sh cilium-1.14.11.tgz
SUCCESS: cilium {$.clustermesh.apiserver.image.tag}=v1.14.11 matches chart version 1.14.11
SUCCESS: cilium {$.hubble.relay.image.tag}=v1.14.11 matches chart version 1.14.11
SUCCESS: cilium {$.image.tag}=v1.14.11 matches chart version 1.14.11
SUCCESS: cilium {$.operator.image.tag}=v1.14.11 matches chart version 1.14.11
SUCCESS: cilium {$.preflight.image.tag}=v1.14.11 matches chart version 1.14.11

% ./validate_helm_chart.sh cilium-1.13.16.tgz
SUCCESS: cilium {$.clustermesh.apiserver.image.tag}=v1.13.16 matches chart version 1.13.16
SUCCESS: cilium {$.hubble.relay.image.tag}=v1.13.16 matches chart version 1.13.16
SUCCESS: cilium {$.image.tag}=v1.13.16 matches chart version 1.13.16
SUCCESS: cilium {$.operator.image.tag}=v1.13.16 matches chart version 1.13.16
SUCCESS: cilium {$.preflight.image.tag}=v1.13.16 matches chart version 1.13.16
```

Reply:
There's probably a way to dig these out of the Helm chart itself:
```
cilium $ grep -R 'Values.*tag' install/kubernetes/ | sed 's/^.*Values\([^ ]*\.tag\).*$/$\1/g'
$.hubble.ui.backend.image.tag
$.hubble.ui.frontend.image.tag
$.hubble.ui.backend.image.tag
$.hubble.ui.frontend.image.tag
$.operator.image.tag
```
...
```
tetragon $ grep -R 'Values.*tag' install/kubernetes/ | sed 's/^.*Values\([^ ]*\.tag\).*$/$\1/g'$.tetragon.image.tag
$.export.stdout.image.tag
$.tetragonOperator.image.tag
$.tetragon.image.tag
```

Maybe pipe into `sort -u` to remove duplicates.

Reply:
Yeah, this can be a good solution!

Reply:
the tricky part is picking which ones to check. for example:
```
$.export.stdout.image.tag
```
it's expected this version does not match tetragon chart version. 

for cilium chart:
```
$.hubble.ui.backend.image.tag
$.hubble.ui.frontend.image.tag
$.hubble.ui.backend.image.tag
$.hubble.ui.frontend.image.tag
```
these don't match cilium chart version, but `$.operator.image.tag` does 🤔💭

Reply:
Yeah I suspected this would be the awkward part. Looking back on the first Cilium command above, it's also missing a bunch of the base images. That said, we may be overthinking this. I'm not exactly sure the danger if we happened to add a new image and forget to add it to a list here. We could always do auto-detect similar to above but `grep -v -e 'stdout' -e 'hubble\.ui'`, or we could just come up with a list and call it a day.

Reply:
let's just say it's a problem to solve in the future. i'm shipping as is 🚢